src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtOpeningTimesController.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.dts.fact.controllers.admin;
+
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.dts.fact.config.security.Role;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.OpeningTime;
+import uk.gov.hmcts.dts.fact.services.admin.AdminCourtOpeningTimeService;
+
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.dts.fact.controllers.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.controllers.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@RestController
+@RequestMapping(
+    path = "courts/{slug}/openingTimes",
+    produces = {MediaType.APPLICATION_JSON_VALUE}
+)
+public class AdminCourtOpeningTimesController {
+    private final AdminCourtOpeningTimeService adminService;
+
+    @Autowired
+    public AdminCourtOpeningTimesController(AdminCourtOpeningTimeService adminService) {
+        this.adminService = adminService;
+    }
+
+    @GetMapping()
+    @ApiOperation("Find court opening times by slug")
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<OpeningTime>> getCourtOpeningTimes(@PathVariable String slug) {
+        return ok(adminService.getCourtOpeningTimesBySlug(slug));
+    }
+
+    @PutMapping()
+    @ApiOperation("Update court opening times")
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<OpeningTime>> updateCourtOpeningTimes(@PathVariable String slug, @RequestBody List<OpeningTime> openingTimes) {
+        return ok(adminService.updateCourtOpeningTimes(slug, openingTimes));
+    }
+
+    @ExceptionHandler(NotFoundException.class)
```

Comment:
We may want to put this into its own ExceptionHandler class in the future? Do we think this can be reused? Not quite sure if stuff will be routed to here from other controllers once springboot maps everything together on startup. One to think about :)

Reply:
Good point Adam. Some other controllers are handling the same exception. I will move it out to be handled globally